### PR TITLE
Fix gradient checkpointing bug M2M 100

### PR DESCRIPTION
**Pull request by KMFODA**
This PR fixes a bug that a user can encounter while using generate and models that use gradient_checkpointing.

Fixes Issue https://github.com/huggingface/transformers/issues/21737

cc @younesbelkada or @gante

### CODE REVIEW

**Review by younesbelkada — COMMENTED** [review]
Hey @KMFODA 
Thanks a mile for this! 
Can you please run the styling checks (`make fixup`), we should be good to go afterwards!

**Review by younesbelkada — APPROVED** [review]
Thanks a lot for this!

File: src/transformers/models/m2m_100/modeling_m2m_100.py [hunk 1]
```diff
@@ -1024,6 +1024,13 @@ def forward(
 
         hidden_states = nn.functional.dropout(hidden_states, p=self.dropout, training=self.training)
 
+        if self.gradient_checkpointing and self.training:
+            if use_cache:
+                logger.warning(
```
Comment (sgugger) [inline src/transformers/models/m2m_100/modeling_m2m_100.py:1029]:
```suggestion
                logger.warning_once(
```

**Review by sgugger — APPROVED** [review]
Thanks for working on this. The warning needs to be updated to go with the latest changes on main.
